app/routes.py

```diff
@@ -1,2 +1,184 @@
+
+@videos_bp.route("/<int:video_id>", methods = ["DELETE"], strict_slashes=False)
+def delete_video(video_id):
+    video = Video.query.get_or_404(video_id)
+    db.session.delete(video)
+    db.session.commit()
+    return jsonify(id=video.id)
+
+@videos_bp.route("/<int:video_id>", methods = ["PUT"], strict_slashes=False)
+def update_video(video_id):
+    video = Video.query.get_or_404(video_id)
+    request_body = request.get_json()
+    if invalid_video_request(request_body):
+        return jsonify(details=f"Invalid data"), 400
+    video.title=request_body["title"]
+    video.release_date=request_body["release_date"]
+    video.total_inventory=request_body["total_inventory"]
+    db.session.commit()
+    return jsonify(video.to_json())
+
+@videos_bp.route("/<int:video_id>/rentals", methods = ["GET"], strict_slashes=False)
+def view_customer_rentals(video_id):
+    video = Video.query.get_or_404(video_id)
+    video_rentals = []
+    for rental in video.customers:
+        customer = Customer.query.get(rental.customer_id)
+        customer_details = {"due_date": rental.due_date,
+                            "name": customer.name,
+                            "phone": customer.phone,
+                            "postal_code": customer.postal_code}
+        video_rentals.append(customer_details)
+    return jsonify(video_rentals)
+
+######################## RENTAL ENDPOINTS ######################## 
+
+@rentals_bp.route("/check-out", methods = ["POST"], strict_slashes=False)
+def checkout():
+    request_body = request.get_json()
+    if invalid_checkout_checkin(request_body):
+        return jsonify(details=f"Invalid data"), 400
+    new_rental = Rental(customer_id = request_body["customer_id"],
+                        video_id = request_body["video_id"])
+    customer = Customer.query.get(request_body["customer_id"])
+    customer.videos_checked_out_count += 1
+    video = Video.query.get(request_body["video_id"])
+    if not video.available_inventory:
+        return jsonify(details=f"Video not available"), 400
+    video.available_inventory -= 1
+    db.session.add(new_rental)
+    db.session.commit()
+    rental_response = new_rental.to_json()
+    rental_response["due_date"] = new_rental.due_date
+    return jsonify(rental_response)
+
+@rentals_bp.route("/check-in", methods = ["POST"], strict_slashes=False)
+def checkin():
+    request_body = request.get_json()    
+    if invalid_checkout_checkin(request_body):
+        return jsonify(details=f"Invalid data"), 400
+    customer = Customer.query.get(request_body["customer_id"])
+    video = Video.query.get(request_body["video_id"])
+    rental = Rental.query.filter_by(video_id=video.id, customer_id=customer.id).first()
+    if rental:
+        customer.videos_checked_out_count -= 1
+        video.available_inventory += 1
+        db.session.delete(rental)
+        db.session.commit()
+        return jsonify(rental.to_json())
+    return jsonify(details=f"Invalid data"), 400
```

Comment:
For consistency and readability, consider first checking if rental is none, and then moving the main logic of the function outside of a conditional.